spec/lib/booking_requests/booking_requests_spec.rb

```diff
@@ -2,8 +2,16 @@
   let(:api) { instance_double(BookingRequests::Api) }
 
   describe '.slots' do
-    before do
+    around do |example|
+      previous_api = BookingRequests.api
       BookingRequests.api = api
+
+      example.run
+
+      BookingRequests.api = previous_api
```

Comment:
I think I've used it in some feature specs and if so I've certainly not been employing `ensure`, so will just double check to see if that's the case.